src/weather_gen.h

```diff
@@ -3,13 +3,82 @@
 #define CATA_SRC_WEATHER_GEN_H
 
 #include <string>
+#include <climits>
 
 #include "calendar.h"
+#include "color.h"
 
 struct tripoint;
 class JsonObject;
+using weather_type = int;
 
-enum weather_type : int;
+enum class precip_class : int {
+    NONE,
+    VERY_LIGHT,
+    LIGHT,
+    HEAVY
+};
+
+enum class sun_intensity_type : int {
+    none,
+    light,
+    normal,
+    high,
+};
+
+enum class time_requirement_type : int {
+    day,
+    night,
+    both
+};
+
+/**
+ * Weather animation class.
+ */
+struct weather_animation_t {
+    float    factor;
+    nc_color color;
+    char     glyph;
+};
+
+struct weather_requirements {
+    int windpower_min = INT_MIN;
+    int windpower_max = INT_MAX;
+    int temperature_min = INT_MIN;
+    int temperature_max = INT_MAX;
+    int pressure_min = INT_MIN;
+    int pressure_max = INT_MAX;
+    int humidity_min = INT_MIN;
+    int humidity_max = INT_MAX;
+    bool humidity_and_pressure = true;
+    bool acidic = false;
+    time_requirement_type time;
+    std::vector<std::string> required_weathers;
+};
+
+using weather_effect_fn = void ( * )( int intensity );
+struct weather_datum {
+    std::string id;               //!< Internal name of weather
```

Comment:
Please use a `string_id` as the type for an id.
```suggestion
    string_id<weather_datum> id;               //!< Internal id of weather
```
Or make a global typedef for it.